api/client/joinservice.go

```diff
@@ -125,3 +131,71 @@ func (c *JoinServiceClient) RegisterUsingAzureMethod(ctx context.Context, challe
 	}
 	return certsResp.Certs, nil
 }
+
+// RegisterUsingTPMMethod registers the caller using the TPM join method and
+// returns signed certs to join the cluster. The caller must provide a
+// ChallengeResponseFunc which returns a *proto.RegisterUsingTPMMethodRequest
+// for a given challenge, or an error.
+func (c *JoinServiceClient) RegisterUsingTPMMethod(
+	ctx context.Context,
+	initReq *proto.RegisterUsingTPMMethodInitialRequest,
+	solveChallenge RegisterTPMChallengeResponseFunc,
+) (*proto.Certs, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	tpmJoinClient, err := c.grpcClient.RegisterUsingTPMMethod(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	err = tpmJoinClient.Send(&proto.RegisterUsingTPMMethodRequest{
+		Payload: &proto.RegisterUsingTPMMethodRequest_Init{
+			Init: initReq,
+		},
+	})
+	if err != nil {
+		return nil, trace.Wrap(err, "sending initial request")
+	}
+
+	challengeResp, err := tpmJoinClient.Recv()
+	if err != nil {
+		return nil, trace.Wrap(err, "receiving challenge")
+	}
+
+	challenge, ok := challengeResp.Payload.(*proto.RegisterUsingTPMMethodResponse_ChallengeRequest)
+	if !ok {
+		return nil, trace.BadParameter(
+			"unexpected payload type %T, expected *RegisterUsingTPMMethodResponse_ChallengeRequest",
+			challengeResp.Payload,
+		)
+	}
+
+	solution, err := solveChallenge(challenge.ChallengeRequest)
+	if err != nil {
+		return nil, trace.Wrap(err, "calling solveChallenge")
```

Comment:
```suggestion
		return nil, trace.Wrap(err, "solving challenge")
```

So it's a bit more user-friendly?